lms/djangoapps/support/views/program_enrollments.py

```diff
@@ -98,3 +106,153 @@ def _validate_and_link(program_uuid_string, linkage_text):
+            return {}, 'Could not find edx account with {}'.format(username_or_email)
+
+    def _get_enrollments(self, user=None, external_user_key=None):
+        """
+        With the user or external_user_key passed in,
+        return a dictionary with corresponding ProgramEnrollments and ProgramCourseEnrollments
+        all serialized for view
+        """
+        program_enrollments = fetch_program_enrollments_by_student(
+            user=user,
+            external_user_key=external_user_key
+        ).prefetch_related('program_course_enrollments')
+
+        enrollments_by_program_uuid = {}
+        for program_enrollment in program_enrollments:
+            serialized_program_enrollment = self._serialize_program_enrollment(program_enrollment)
+            enrollment_item = {
+                'program_enrollment': serialized_program_enrollment
+            }
+            program_course_enrollments = program_enrollment.program_course_enrollments.all()
+            for program_course_enrollment in program_course_enrollments.select_related(
+                'course_enrollment'
+            ):
+                serialized_program_course_enrollment = self._serialize_program_course_enrollment(
+                    program_course_enrollment
+                )
+                enrollment_item.setdefault('program_course_enrollments', []).append(
+                    serialized_program_course_enrollment
+                )
+            enrollments_by_program_uuid[program_enrollment.program_uuid] = enrollment_item
+
+        return list(enrollments_by_program_uuid.values())
+
+    def _serialize_program_enrollment(self, program_enrollment):
+        if not program_enrollment:
+            return {}
+
+        return {
+            'created': program_enrollment.created.strftime(DATETIME_FORMAT),
+            'modified': program_enrollment.modified.strftime(DATETIME_FORMAT),
+            'program_uuid': str(program_enrollment.program_uuid),
+            'external_user_key': program_enrollment.external_user_key,
+            'status': program_enrollment.status
+        }
+
+    def _serialize_program_course_enrollment(self, program_course_enrollment):
+        """
+        Return a dictionary of ProgramCourseEnrollment serialized
+        """
+        if not program_course_enrollment:
+            return {}
+
+        course_enrollment = program_course_enrollment.course_enrollment
+        return {
+            'created': program_course_enrollment.created.strftime(DATETIME_FORMAT),
+            'modified': program_course_enrollment.modified.strftime(DATETIME_FORMAT),
+            'course_enrollment': {
+                'course_id': str(course_enrollment.course_id),
+                'is_active': course_enrollment.is_active,
+                'mode': course_enrollment.mode,
+            },
+            'status': program_course_enrollment.status,
+            'course_key': str(program_course_enrollment.course_key),
+        }
```

Comment:
You're essentially doing the work that DRF serializers already do. Unless it'd add a bunch of complexity, consider using them.

Reply:
Will try to use serializer in my Part 2 PR for all fo these. Thank you for this suggestion!